### DISCUSSION TITLE
What privileges does a privileged container hold?

### DISCUSSION THREAD

**Discussion by kimo6416337** [moby/moby#44002] [Q&A]
Hi all,

I am wondering what privileges are assigned to a privileged container.

After seeing the source code in [https://github.com/moby/moby/blob/master/daemon/oci_linux.go](https://github.com/moby/moby/blob/master/daemon/oci_linux.go) and [https://github.com/moby/moby/blob/master/daemon/seccomp_linux.go](https://github.com/moby/moby/blob/master/daemon/seccomp_linux.go), I think a privileged container at least has:
1. unconfined AppArmor/seccomp profile.
2. mount the host's /sys as rw.
3. grant all linux capabilities.
4. maps all host's devices with rmw (--device /dev/xxx:rmw)

What I want to ask is
1. Is my thought correct? pls correct it.
2. Am I missing anything else? pls mention it.

Thanks!

**Comment by thaJeztah** [comment]
Yes, I think that is about correct. With all of the above, the container effectively won't provide any protection (it will have enough privileges to gain the equivalent of `root` access on the host).

**Comment by thediveo** [comment]
also several procfs overmounts aren't done in case of privileged. You can check the exact overmounts with a tool like my [lxkns service](https://github.com/thediveo/lxkns) in the mount namespaces view.
